engine/src/main/java/org/entando/entando/aps/system/services/page/serializer/WidgetConfigPropertiesDeserializer.java

```diff
@@ -0,0 +1,68 @@
+package org.entando.entando.aps.system.services.page.serializer;
+
+import com.fasterxml.jackson.databind.util.StdConverter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class WidgetConfigPropertiesDeserializer extends StdConverter<Map<String, Object>, Map<String, Object>> {
+
+    @Override
+    public Map<String, Object> convert(Map<String, Object> json) {
+        Map<String, Object> properties = new HashMap<>();
+
+        for (Entry<String, Object> property : json.entrySet()) {
+            if (property.getKey().equals("categories")) {
+                properties.put(property.getKey(), readCategories((List<String>) property.getValue()));
```

Comment:
Add logging for these overrides so that it is clear that we're impacting the serialized object and a dev can see if something comes out unexpected.